test/browser/http.test.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+define(['ably', 'shared_helper', 'chai'], function (Ably, helper, chai) {
+  var rest;
+  var expect = chai.expect;
+
+  describe('rest/http/fetch', function () {
+    this.timeout(60 * 1000);
+    before(function (done) {
+      Ably.Rest.Platform.Config.xhrSupported = false;
+      Ably.Rest.Platform.Config.jsonpSupported = false;
+      helper.setupApp(function () {
+        rest = helper.AblyRest({
+          agents: {
+            'custom-agent': '0.1.2',
+          },
+        });
+        done();
+      });
+    });
+
+    after((done) => {
+      Ably.Rest.Platform.Config.xhrSupported = window.XMLHttpRequest && 'withCredentials' in new XMLHttpRequest();
```

Comment:
It would be better to store a reference to the old values before you overwrite them, rather than duplicating the logic here

Reply:
https://github.com/ably/ably-js/pull/1019/commits/07ddaba5a65cbbe808fba37c6d46a3c0eb511cda